python/observation-publisher/publisher/__main__.py

```diff
@@ -12,11 +13,18 @@ def start_consumer(cfg: PublisherConfig) -> None:
     if missing_keys:
         raise RuntimeError(f"Got missing keys in config:\n{missing_keys}")
     prediction_log_consumer = new_consumer(cfg.environment.observation_source)
+    inference_schema = InferenceSchema.from_dict(
```

Comment:
where this `from_dict` method is declared? is it part of dataclass method?

Reply:
This comes from the `@dataclass_json` annotation. The decorator modifies the original class, add additional methods, then register the original class as a virtual subclass to DataclassJsonMixIn.